src/abstractdataframe/reshape.jl

```diff
@@ -437,17 +452,28 @@ mutable struct RepeatedVector{T} <: AbstractVector{T}
     outer::Int
 end
 
-function Base.getindex(v::RepeatedVector{T},i::AbstractVector{I}) where {T,I<:Real}
-    N = length(v.parent)
-    idx = Int[Base.fld1(mod1(j,v.inner*N),v.inner) for j in i]
-    v.parent[idx]
+function Base.getindex(v::RepeatedVector, i::Bool)
+    Base.depwarn("Indexing RepeatedVector with Bool is deprecated", :getindex)
+    v[1]
+end
+
+function Base.getindex(v::RepeatedVector, i::Real)
+    Base.depwarn("Indexing RepeatedVector with Real is deprecated", :getindex)
+    v[Int(i)]
 end
-function Base.getindex(v::RepeatedVector{T},i::Real) where T
+
+function Base.getindex(v::RepeatedVector, i::Integer)
     N = length(v.parent)
     idx = Base.fld1(mod1(i,v.inner*N),v.inner)
     v.parent[idx]
 end
-Base.getindex(v::RepeatedVector,i::AbstractRange) = getindex(v, [i;])
+
+Base.getindex(v::RepeatedVector,i::AbstractVector) = [v[j] for j in i]
```

Comment:
I'm surprised there is no fallback doing the same thing already.

Reply:
Good point. Should not be necessary.